source/client.c

```diff
@@ -561,6 +621,25 @@ void aws_mqtt_client_connection_destroy(struct aws_mqtt_client_connection *conne
  * Connection Configuration
  ******************************************************************************/
 
+/* To configure the connection, ensure the state is DISCONNECTED or CONNECTED */
+static int s_check_connection_state_for_configuration(struct aws_mqtt_client_connection *connection) {
+    int result = AWS_OP_SUCCESS;
+    { /* BEGIN CRITICAL SECTION */
+        s_mqtt_connection_lock_synced_data(connection);
+
+        if (connection->synced_data.state != AWS_MQTT_CLIENT_STATE_DISCONNECTED &&
+            connection->synced_data.state != AWS_MQTT_CLIENT_STATE_CONNECTED) {
+            AWS_LOGF_WARN(
+                AWS_LS_MQTT_CLIENT,
+                "id=%p: The connection is not stable now. Not safe to configure the connection. Try again later.",
```

Comment:
"Connection is currently pending connect/disconnect. Unable to make configuration changes until pending operation completes"